.github/workflows/buf.yaml

```diff
@@ -0,0 +1,34 @@
+name: Buf
+
+on:
+  pull_request:
+
+permissions:
+  contents: read
+  pull-requests: read
+
+jobs:
+  lint-protos:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: bufbuild/buf-setup-action@v1
+      - uses: bufbuild/buf-lint-action@v1
+  generate-protos:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.ref }}
+          repository: ${{ github.event.pull_request.head.repo.full_name }}
+      - run: make proto
+      - uses: EndBug/add-and-commit@v9
+        id: commit
+        with:
+          add: 'gen'
+          default_author: github_actions
+          message: 'fix: generate protobufs'
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
If the output of `make proto` is different from what was committed, the ci must fail. When this is the case, a failed `generate-protos / validate-protos` ci indicated that the application was likely developed with a different version of the protos.